### PR TITLE
Work around issue with doxygen config

### LINKED ISSUE #338: Conan fails to install doxygenconfig.cmake on GHA runners
Same issue found on @tkoskela Ubuntu 22.04 laptop with conan versions 1.54 -- 1.58. No `doxygenconfig.cmake` in build directory after `conan install`.

Works correctly on @SJaffa Ubuntu 20.04 laptop with conan versions 1.51 -- 1.58.

### PR DESCRIPTION
**Pull request by tkoskela**

### CODE REVIEW

**Comment by tkoskela** [comment]
This is a temporary solution, we should wait for some feedback from conan developers in https://github.com/conan-io/conan/issues/13149 before merging.

- Installing a fixed revision of doxygen through conan in a separate `conan install` command seems to work, while installing it through `conanfile.py` did not.
- We are installing the second-newest revision of the doxygen package, the newst one does not seem to work.
- doxygen does not get added to the global path and we couldn't easily retrieve its path, therefore it is also installed by `apt` to use in the Make Docweb step. The version in `apt` is much older, but does not seem to break things.

- We made changes to the conanfile suggested by comments in https://github.com/conan-io/conan/issues/13149. They are in line with changes that will be necessary to migrate to conan 2.0.0. However, they don't seem to directly influence the issue with doxygen config files not being written that was fixed by conan installing doxygen separately.

**Review by SJaffa — APPROVED** [review]

**Pull request merged**
